### ISSUE TITLE
Consider removing fat LTO

### ISSUE THREAD

**Issue by Nashtare** [0xPolygonZero/zero-bin#71]
We currently compile the `release` profile with fat link-time optimization, which hinders pretty harshly compile time with apparently non clearly visible runtime performance gains.

If this is confirmed, we may consider removing this for faster compilation / re-compilation.

**Comment by BGluth** [comment]
Hmm it's still nice to be able to optimize across library calls, and I would be surprised if performance is actually worse with this, unless the compiler tries to optimize is a bad way. We do need to do `release` builds for testing, so removing this from `release` makes sense.

Just an idea, but wdyt about adding a second profile (called `hardcore-release` or something :sweat_smile:) to the crate that enables this and any other "non-essential" optimizations? It would still be nice to build with link time optimization for binaries that people intend to use for production, and we're in a bit of an unusual spot I think where `debug` builds don't cut it for testing.

**Comment by Nashtare** [comment]
Yeah we could definitely have some debug / testing / release profiles. Though from preliminary and incomplete benchmarks, it didn't seem like LTO was bringing any significant improvement, while heavily hindering compile time
